### PR TITLE
Adds CLI for gradient-between-adjacent-grid-squares

### PR DESCRIPTION
**Pull request by MoseleyS**
Addresses https://metoffice.atlassian.net/browse/EPPT-1162

Adds a CLI for the GradientBetweenAdjacentGridSquares plugin. While setting up tests for this, I found it needed to handle "status_flag" AuxCoords too, so I have added this.

Acceptance test data: https://github.com/metoppv/improver_test_data/pull/56

Testing:
 - [x] Ran tests and they passed OK
 - [x] Added new tests for the new feature(s)

### CODE REVIEW

**Review by mspelman07 — CHANGES_REQUESTED** [review]
Broadly does what we need but a few tweaks for you.

File: improver/cli/gradient_between_adjacent_grid_squares.py [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+"""CLI to calculate the gradient between adjacent grid squares."""
+from improver import cli
+
+
+@cli.clizefy
+@cli.with_output
+def process(cube: cli.inputcube, *, regrid: bool = False):
+    """Module to calculate the size of hail stones from the
+    cloud condensation level (ccl) temperature and pressure, temperature
+    on pressure levels data, wet bulb freezing altitude above sea level and orography.
```
[the remark this answers is not in the document]
Comment (MoseleyS) [reply improver/cli/gradient_between_adjacent_grid_squares.py:15]: Fixed

File: improver/cli/gradient_between_adjacent_grid_squares.py [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+        cube (iris.cube.Cube):
+            Cube from which the differences will be calculated.
+        regrid:
+                If True, the gradient cube is regridded to match the spatial
+                dimensions of the input cube. If False, the two output gradient cubes will have
+                different spatial coords such that the coord matching the gradient axis will
+                represent the midpoint of the input cube and will have one fewer points.
+                If the x-axis is marked as circular, the gradient between the last and first points
+                is also included.
+                If a status_flag coord is present, it will be preserved if regrid is true.
```
[the remark this answers is not in the document]
Comment (MoseleyS) [reply improver/cli/gradient_between_adjacent_grid_squares.py:27]: Done

File: improver/cli/gradient_between_adjacent_grid_squares.py [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+@cli.clizefy
+@cli.with_output
+def process(cube: cli.inputcube, *, regrid: bool = False):
+    """Module to calculate the size of hail stones from the
+    cloud condensation level (ccl) temperature and pressure, temperature
+    on pressure levels data, wet bulb freezing altitude above sea level and orography.
+
+    Args:
+        cube (iris.cube.Cube):
+            Cube from which the differences will be calculated.
```
[the remark this answers is not in the document]
Comment (MoseleyS) [reply improver/cli/gradient_between_adjacent_grid_squares.py:19]: Corrected

File: improver/cli/gradient_between_adjacent_grid_squares.py [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+                dimensions of the input cube. If False, the two output gradient cubes will have
+                different spatial coords such that the coord matching the gradient axis will
+                represent the midpoint of the input cube and will have one fewer points.
+                If the x-axis is marked as circular, the gradient between the last and first points
+                is also included.
+                If a status_flag coord is present, it will be preserved if regrid is true.
+
+    Returns:
+        Cube after the gradients have been calculated along the x-axis and Cube after the gradients
+        have been calculated along the y-axis.
```
[the remark this answers is not in the document]
Comment (MoseleyS) [reply improver/cli/gradient_between_adjacent_grid_squares.py:31]: I've decided not to change this one, I think it is sufficiently described already.

File: improver/utilities/spatial.py [hunk 5]
```diff
@@ -682,6 +684,13 @@ def _create_output_cube(gradient: Cube, name: str) -> Cube:
         )
         return grad_cube
 
+    def _strip_status_flag(self, cube: Cube):
+        """Finds and removes any status_flag coord as this isn't handled by sub-Plugins"""
+        flag_coords = [c for c in cube.coords() if "status_flag" in c.name()]
+        if flag_coords:
+            (self.status_flag_coord,) = flag_coords
+            cube.remove_coord(self.status_flag_coord)
```
[the remark this answers is not in the document]
Comment (MoseleyS) [reply improver/utilities/spatial.py:692]: Standardise will remove a "air_temperature status_flag" coord only. Your suggestion is valid but I'm not sure that IMPROVER has a consistent method for handling these.

[the remark this answers is not in the document]
Comment (mspelman07) [reply improver/utilities/spatial.py:692]: I think following our conversation we've decided to strip the status flag coord in the standardise chain and put it back in at the end so the function and tests can have the reference to the status flag removed.

File: improver/cli/gradient_between_adjacent_grid_squares.py (hunk 4 above)
[the remark this answers is not in the document]
Comment (MoseleyS) [reply improver/cli/gradient_between_adjacent_grid_squares.py]: While attempting to fix Sphinx, I've found that you were right, so this is now declared as being a tuple of two Cubes.

File: improver/utilities/spatial.py (hunk 5 above)
[the remark this answers is not in the document]
Comment (MoseleyS) [reply improver/utilities/spatial.py]: Done

File: improver/cli/gradient_between_adjacent_grid_squares.py [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+def process(cube: cli.inputcube, *, regrid: bool = False):
+    """Calculate the gradient between adjacent grid squares within
+    a cube. The gradient is calculated along the x and y axis
+    individually.
+
+    Args:
+        cube (iris.cube.Cube):
+            Cube from which the gradients will be calculated.
+        regrid (bool):
+                If True, the gradient cube is regridded to match the spatial
```
Comment (mspelman07) [inline improver/cli/gradient_between_adjacent_grid_squares.py:21]: The indentation here has gone strange again. Is this something black/flake is doing?

**Review by mspelman07 — COMMENTED** [review]
One tiny thing but otherwise I'm happy

File: improver/cli/gradient_between_adjacent_grid_squares.py (hunk 6 above)
Comment (MoseleyS) [reply improver/cli/gradient_between_adjacent_grid_squares.py]: I suspect it is just me getting it wrong

**Review by mspelman07 — APPROVED** [review]

**Review by seshukolusu16 — APPROVED** [review]

**Pull request merged**